### PR TITLE
Don't show great or higher judgements when using argon "pro" skin

### PR DESCRIPTION
**Pull request by peppy**
Done in the easiest-to-implement way for now. I would have liked to use `MaxResult` from the triggering `Judgement` but doing so would come with its own weirdness due to how drawable judgements are pooled. Eventually we will want to move this logic inside `DrawableJudgement` to allow for toggling as a setting, but that will come later.

The base class structural changes aren't strictly required for this PR, but I've included here because it's a pretty minor change (will be built on in upcoming changes).

### CODE REVIEW

File: osu.Game/Rulesets/Judgements/JudgementPiece.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        protected JudgementPiece(HitResult result)
+        {
+            Result = result;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            JudgementText.Text = Result.GetDescription().ToUpperInvariant();
```
Comment (bdach) [inline osu.Game/Rulesets/Judgements/JudgementPiece.cs:26]: This is a bit dodgy... Basically all implementors of this are expected to know that they should (a) create `JudgementText` themselves, and (b) do so before `base.LoadComplete()`. Is this needed? Can we expose just the `LocalisableString` with the text or something? Or move the creation of the judgement text to an abstract method that you can't not implement?

File: osu.Game.Rulesets.Mania/Skinning/Argon/ManiaArgonSkinTransformer.cs [hunk 2]
```diff
@@ -27,6 +27,10 @@ public ManiaArgonSkinTransformer(ISkin skin, IBeatmap beatmap)
             switch (lookup)
             {
                 case GameplaySkinComponentLookup<HitResult> resultComponent:
+                    // This should eventually be moved to a skin setting, when supported.
+                    if (Skin is ArgonProSkin && resultComponent.Component >= HitResult.Great)
```
Comment (bdach) [inline osu.Game.Rulesets.Mania/Skinning/Argon/ManiaArgonSkinTransformer.cs:31]: Just to confirm intent - did this mean to use `HitResult.Great` in mania specifically (as opposed to `HitResult.Perfect`)?

Comment (peppy) [reply osu.Game.Rulesets.Mania/Skinning/Argon/ManiaArgonSkinTransformer.cs:31]: Will need player feedback on this, but `Great` is still generally considered "high enough" to not be shown it, I believe. At least in stable they don't have a detrimental effect on your score.

File: osu.Game/Rulesets/Judgements/JudgementPiece.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Rulesets/Judgements/JudgementPiece.cs:26]: I had more in that direction on a separate branch, will move it over here.

**Comment by bdach** [comment]
I've fixed a code inspection (https://github.com/ppy/osu/pull/21774/commits/f973befcd4f0f24701917117b5a9fe41d345b7fe), but also, more importantly, fixed the new `JudgementText` in the base class never being added to the judgement (https://github.com/ppy/osu/pull/21774/commits/80de5dac66776d9da6348ff60421bb1e0c66a49f). Looks okay to me now, but please double check the changes.

**Review by bdach — APPROVED** [review]

File: osu.Game.Rulesets.Mania/Skinning/Argon/ManiaArgonSkinTransformer.cs (hunk 2 above)
Comment (0xallie) [reply osu.Game.Rulesets.Mania/Skinning/Argon/ManiaArgonSkinTransformer.cs:31]: Now that lazer is using ScoreV2 perhaps this should be reevaluated, as Great now reduces accuracy in mania compared to Perfect.

**Pull request merged**